syz-fuzzer/workqueue.go

```diff
@@ -129,3 +125,22 @@ func (wq *WorkQueue) wantCandidates() bool {
 	defer wq.mu.RUnlock()
 	return len(wq.candidate) < wq.procs
 }
+
+// We might have used a heap for this, but the cost of traversing the queue
+// is miniscule compared to the cost of executing a program.
+func popWorkTriage(queue []*WorkTriage) ([]*WorkTriage, *WorkTriage) {
+	var retIdx int
+	newQueue := make([]*WorkTriage, len(queue)-1)
+	for i, item := range queue {
+		if len(queue[retIdx].info.Signal) < len(item.info.Signal) {
+			retIdx = i
+		}
+	}
+	if retIdx > 0 {
```

Comment:
This if can be removed. The next one probably can be removed with some code restructuring as well.

But this really asks for a priority queue:
https://pkg.go.dev/container/heap#example__priorityQueue

The current version generated quadratic garbage and quadratic memory copying.

If you don't want real priority queue, a map be simpler and faster (supports delete).